x/distribution/keeper/keeper.go

```diff
@@ -231,6 +231,10 @@ func (k Keeper) WithdrawSingleShareRecordReward(ctx context.Context, recordID ui
 	}
 	owner := sdk.AccAddress(ownerAddr)
 
+	if k.bankKeeper.BlockedAddr(owner) {
```

Comment:
@stana-miric Could you move the checks before the `SendCoins` calls? It will make it easier to understand later why they are necessary.

Reply:
That would do unnecessary work since there's a bunch of store access that happens before the send invocation.

What do you thing about just adding a comment annotation explaining why?

(Personal opinion: functions that exit as early as possible using guards are a lot easier to read and understand).

Reply:
Sound good.